simulate/simulate.go

```diff
@@ -236,9 +237,16 @@ func MutateGene(inputSeq []dna.Base, branchLength float64, geneFile string, dele
 
 	}
 	mutatedSequence := BaseExtToBases(newSequence)
-	delFound := seqContainsDels(mutatedSequence, dna.Gap)
+	delFound := seqContainsDels(mutatedSequence)
 	if delFound == false && deletions == true {
-		finalSequence = deleteBase(mutatedSequence)
+		delEvent := rand.Float64()
+		switch {
+		case delEvent <= branchLength :
+			finalSequence = deleteBase(mutatedSequence)
+		default:
+			finalSequence = mutatedSequence
+		}
```

Comment:
@craiglowe here is a potential solution to your suggestion of making deletion probabilities a function of the branch length. I made the assumption that branch lengths are typically between 0 and 1 so any branches longer than this should always result in the addition of a deletion.